coderd/tailnet_test.go

```diff
@@ -389,9 +426,36 @@ type agentWithID struct {
 	agent.Agent
 }
 
-func setupServerTailnetAgent(t *testing.T, agentNum int, opts ...tailnettest.DERPAndStunOption) ([]agentWithID, *coderd.ServerTailnet) {
```

Comment:
Self-review: had to refactor this a bit to allow existing options to be passed in or a healthcheck func. It's a bit janky but it keeps the scope of changes tight; open to other ideas.